File: leetcode/hash tables/383. Ransom Note/golang/main.go
package main

import "fmt"

func canConstruct(ransomNote string, magazine string) bool {
	counts := make(map[rune]int)

	for _, char := range magazine {
		counts[char]++
	}

	for _, char := range ransomNote {
		counts[char]--
		if counts[char] < 0 {
			return false
		}
	}

	return true
}
func main() {
	ransomNote := "aa"
	magazine := "aab"
	fmt.Println(canConstruct(ransomNote, magazine))
}

File: leetcode/hash tables/383. Ransom Note/python/main.py
from typing import Counter


class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        rnDict, mgDict = Counter(ransomNote), Counter(magazine)

        for k, v in rnDict.items():
            if v > mgDict[k]:
                return False

        return True


print(Solution().canConstruct(ransomNote="aa", magazine="aab"))
